### PR TITLE
docs/best_practices: all contained vault passwords required when using inventory

### PR DESCRIPTION
**Pull request by azrdev**
especially relevant for AWX, which *is not compatible* with the
documented ansible best practice:

<https://github.com/ansible/awx/issues/223#issuecomment-768386089>
<https://github.com/ansible/awx/issues/223#issuecomment-900050306>

##### ISSUE TYPE

- Docs Pull Request

### CODE REVIEW

File: docs/docsite/rst/user_guide/playbooks_best_practices.rst [hunk 1]
```diff
@@ -68,14 +68,14 @@ A system can be in multiple groups.  See :ref:`intro_inventory` and :ref:`intro_
 Separate production and staging inventory
 -----------------------------------------
 
-You can keep your production environment separate from development, test, and staging environments by using separate inventory files or directories for each environment. This way you pick with -i what you are targeting. Keeping all your environments in one file can lead to surprises!
+You can keep your production environment separate from development, test, and staging environments by using separate inventory files or directories for each environment. This way you pick with -i what you are targeting. Keeping all your environments in one file can lead to surprises! For example, all vault passwords used in an inventory need to be available when using that inventory, in other words: Your developers would be able to access production secrets when these environments are kept in the same inventory.
```
Comment (ariordan-redhat) [inline docs/docsite/rst/user_guide/playbooks_best_practices.rst]:
```suggestion
You can keep your production environment separate from development, test, and staging environments by using separate inventory files or directories for each environment. This way you pick with -i what you are targeting. Keeping all your environments in one file can lead to surprises! For example, all vault passwords used in an inventory need to be available when using that inventory. If an inventory contains both production and development environments, developers using that inventory would be able to access production secrets.
```

**Review by ariordan-redhat — COMMENTED** [review]
Suggested rephrasing

**Comment by samccann** [comment]
Thanks @azrdev for the Ansible docs PR!

**Pull request merged**